tests/unit/plugins/test_catkin.py

```diff
@@ -1683,24 +1617,19 @@ def test_find_system_dependencies_system_only_no_packages(self):
         self.catkin_mock.find.assert_called_once_with("bar")
 
     def test_find_system_dependencies_already_satisfied_in_source(self):
-        self.rospack_mock.list_names.return_value = {"bar"}
-
         self.assertThat(
-            catkin._find_system_dependencies(
-                None, self.rosdep_mock, self.rospack_mock, self.catkin_mock
-            ),
+            catkin._find_system_dependencies(None, self.rosdep_mock, self.catkin_mock),
             Equals(dict()),
         )
 
         self.rosdep_mock.get_dependencies.assert_called_once_with()
-        self.rospack_mock.list_names.assert_called_once_with()
-        self.rosdep_mock.resolve_dependency.assert_not_called()
-        self.catkin_mock.find.assert_not_called()
+        self.rosdep_mock.resolve_dependency.assert_called_once_with("bar")
```

Comment:
This is no longer testing the same thing. Is it any different now than [test_find_system_dependencies_system_only_no_packages](https://github.com/snapcore/snapcraft/pull/2833/files#diff-e65da2ca12f490f5cab4eeedc84e6636R1607)? Looks the same to me, but it's missing the mock to make it actually return anything other than an empty dict. I think this entire test should be removed now, as the behavior is only present now within rosdep itself, the calling of which is already well-tested, and the spread test ensures this functionality still exists.

Reply:
Removed this test in d823c5b